x-pack/plugin/esql/compute/gen/src/main/java/org/elasticsearch/compute/gen/AggregatorImplementer.java

```diff
@@ -353,14 +353,14 @@ private MethodSpec addRawInput() {
             builder.addStatement("return");
             builder.endControlFlow();
         }
-        builder.beginControlFlow("if (mask.isConstant())");
+        builder.beginControlFlow("if (mask.allFalse())");
+        {
+            builder.addComment("Entire page masked away");
+            builder.addStatement("return");
+        }
+        builder.endControlFlow();
+        builder.beginControlFlow("if (mask.allTrue())");
```

Comment:
I'm not sure this makes much sense as I understand it (?)

Before, we checked the mask in O(1), if it was all true or not. If it was all true, then we avoided checking the mask by calling `addRawVector(block)`.

Now, we check the mask in O(N), _just to avoid reading the mask later_ (Which is what we do in `addRawVector(block, mask)`). So, 2 possibilities:
1. We read every element of the mask, and it's all true. Then, we did exactly the same as in `addRawVector(block, mask)`
2. We short-circuit by finding a false. Then we're iterating the mask twice (Here and in `addRawVector`)

TL;DR: I would proably switch to `isConstant()` again (At least until we keep some constant within the blocks or something to get allTrue/False in constant time?). I don't think it's a big deal, but we're losing the improvement those ifs gave us

Reply:
This clearly conflicts with the benchmarks you posted. I wonder what do you think about this, or where I'm wrong here 👀

Reply:
Crucially, `isConstant` is allowed to lie. It can return `false` if something is a constant. So it's possible that it'll all be `false` in `addRawVector(vector, mask)`. In that case we can't set `seen` to true up front - we have to set it only when we see a non-`false` value. Which costs. This checking twice avoids that.

Have a look at the commit [99bcb6a](https://github.com/elastic/elasticsearch/pull/113804/commits/99bcb6a74568cc8a90395a5d82421cad117e4c60) - that's how you have to do it if you don't have a function that can't lie.

Another option, I suppose, is for `isConstant` not to ever lie for booleans. Not hard to implement, but I'm worried that it'd cost too much at build time. And I'm not super happy with making `isConstant` do something different for booleans than for numbers. And it's quite important that numbers cal lie about being a constant. It isn't worth checking on them.

About the performance - if the block *is* a constant we get this in O(1) time still. Which is the normal case here. If you aren't using filtering you throw in a constant true block.

Reply:
> Crucially, isConstant is allowed to lie. It can return false if something is a constant.

I suppose by lying you mean "blocks with all the same value, that aren't a `Constant*Vector`". If that's it, indeed. Actually, the method name `isConstant` doesn't quite express that, so I guess +1 to not using it and having more semantic methods.

> In that case we can't set seen to true up front - we have to set it only when we see a non-false value.

Gotcha!

Would be nice to have a continuation or issue to improve the allTrue/False method on non-constant blocks. Not sure if worth it tho, but something to check

Reply:
> Actually, the method name `isConstant` doesn't quite express that, so I guess +1 to not using it and having more semantic methods.

I could rename it in the follow up to `probablyConstant`.

Reply:
To me, it's the "constant" part. What is a "constant" here? From what I understand, a more _semantic_ name would be "alllValuesAreTheSame()", but that's the worst method name ever. "isConstant" is a technical name, and is technically correct, as we have "Constant" blocks.
If a block was created as non-constant for _reasons_, even if all values are the same, it's not lying really. It's not a "_Constant*_" block. 
_(Anyway, philosophy, I wouldn't change anything here now)_

Reply:
Good names are important! I'm not going to change that method in this PR. But we can change it in another.